types/names.go

```diff
@@ -0,0 +1,48 @@
+package types
+
+import (
+	"regexp"
+)
+
+var (
+	MinNameRegistrationPeriod uint64 = 5
+
+	// cost for storing a name for a block is
+	// CostPerBlock*CostPerByte*(len(data) + 32)
+	NameCostPerByte  uint64 = 1
+	NameCostPerBlock uint64 = 1
+
+	MaxNameLength = 32
+	MaxDataLength = 1 << 16
+
+	// Name should be alphanum, underscore, slash
+	// Data should be anything permitted in JSON
+	regexpAlphaNum = regexp.MustCompile("^[a-zA-Z0-9_/]*$")
```

Comment:
For names, the current min is 1 (https://github.com/tendermint/tendermint/blob/namereg/types/tx.go#L202). Data can be empty but maybe thats bad?